### PR TITLE
implemented friend API

### PR DESCRIPTION
**Pull request by HansWongDH**
implemented an friend relationship api to keep track of user to user friend status.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,6 +20,6 @@ fclean:
 	$(COMPOSE) $(DOWN) $(CLEAN_IMAGES) $(CLEAN_ORPHANS) $(CLEAN_VOLUMES)
 	$(DOCKER) system prune -f
 
-re:	fclean hans
+re:	fclean all
```
Comment (42EuLee) [inline Makefile:24]: Very good, Exceptional

File: backend/src/app.module.ts [hunk 2]
```diff
@@ -4,9 +4,10 @@ import { AppService } from './app.service';
 import { DatabaseModule } from './database/database.module';
 import { UsersModule } from './users/users.module';
 import { MatchHistoryModule } from './match_history/match_history.module';
+import { FriendsModule } from './friends/friends.module';
 
 @Module({
```
Comment (42EuLee) [inline backend/src/app.module.ts:9]: Exceptional

File: backend/src/database/database.module.ts [hunk 3]
```diff
@@ -11,7 +13,7 @@ import { User } from '../users/entities/user.entity';
       username: 'admin',
       password: 'admin',
       database: 'ft_transcendence',
```
Comment (42EuLee) [inline backend/src/database/database.module.ts:15]: Obsolutely Breathtaking

File: backend/src/friends/dto/create-friend.dto.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { FriendStatus } from "../entities/friend.entity";
+
+
```
Comment (42EuLee) [inline backend/src/friends/dto/create-friend.dto.ts:3]: Wai you so pro

File: backend/src/friends/entities/friend.entity.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+export enum FriendStatus {
+  Friend = "friend",
+  Invited = "invited",
+  Pending = "pending",
+  Blocked = "blocked",
+  Accept = "accept",
+  Deny = "deny",
+}
+
```
Comment (42EuLee) [inline backend/src/friends/entities/friend.entity.ts:11]: I would have never figure how to do it, thanks for carrying

File: backend/src/friends/entities/friend.entity.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+@Entity()
+export class Friend {
+  @PrimaryGeneratedColumn()
+  uid: number;
+
+  @Column()
+  p1_uid: number;
+
+  @Column()
```
Comment (42EuLee) [inline backend/src/friends/entities/friend.entity.ts:20]: The best entity Is Ding's entity, himself

File: backend/src/friends/friends.controller.ts [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+  @Post()
+  async create(@Query('uid_1', ParseIntPipe) uid_1: number,
+    @Query('uid_2', ParseIntPipe) uid_2: number): Promise<void> {
+    // console.log(senderFriendDto);
+    // console.log(receiverFriendDto);
+    await this.friendsService.create(uid_1, uid_2);
+  }
+
+
```
Comment (42EuLee) [inline backend/src/friends/friends.controller.ts:31]: 5 years worth of code

File: backend/src/friends/friends.service.ts [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    @InjectRepository(Friend)
+    private friendRepository: Repository<Friend>,
+  ) { }
+
+  async create(uid_1: number, uid_2: number): Promise<void> {
+    const senderFriendDto = new CreateFriendDto();
+    senderFriendDto.p1_uid = uid_1;
+    senderFriendDto.p2_uid = uid_2;
+    senderFriendDto.status = FriendStatus.Invited;
+
```
Comment (42EuLee) [inline backend/src/friends/friends.service.ts:21]: Awarded with the best coder award

**Review by hooyunzhe — COMMENTED** [review]
A case of ManyToMany relationship where both tables are the same. Will be interesting to see how we can migrate to using relational decorators, which will be more convenient.

**Review by lokeulee — COMMENTED** [review]
Amazing, exceptional, Bravo, magnifico, Al-dente, toto, Wai you so pro

**Pull request merged**